optaplanner-core/src/main/java/org/optaplanner/core/api/score/buildin/bendable/BendableScoreHolder.java

```diff
@@ -30,7 +30,9 @@
 
 /**
  * @see BendableScore
+ * @deprecated to be replaced by an interface of the same name, removing many methods
  */
+@Deprecated
 public class BendableScoreHolder extends AbstractScoreHolder<BendableScore> {
```

Comment:
I don't think we can deprecate the actual ScoreHolders, like this one. As that means we're forcing people in scoreDRL to use deprecated code, with no alternative to migrate to.

What we can do is deprecate the methods we don't want them to use, and once we branch 8.0, turn BendableScoreHolder etc into an interface with the exact same full class name while only remaining the methods that we allow them to use.